Update license-and-subscription.rst (#4261)
mattermost/docs: source/overview/license-and-subscription.rst

Renewing an Enterprise license
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> You can renew your Mattermost license with a credit card if you have a standard Mattermost contract. When you renew your license, you can also increase the number of active users.

Reviewer: I believe Cloud customers are charged automatically (auto-renew), and hence do not need to go through this process. We should document this in the FAQ.

Author: Added to the FAQ. 👍

Reviewer: My apologies, re-reading the update, I think it makes sense to mention it here, to clarify the renewal.

Something like

```
You can renew your self-managed Mattermost Enterprise license with a credit card if you have a standard Mattermost contract. When you renew your license, you can also increase the number of active users. Y

If you use Mattermost Cloud, your subscription auto-renews so you don't need to follow this process.
```

> If you are a reseller, have a non-standard contract, or want to adjust the number of active users on your license please contact sales@mattermost.com.

Reviewer 2: Question: Is there value in making this email address clickable?

Author: I don't see why not. :)

> If you are a reseller, have a non-standard contract, or want to adjust the number of active users on your license please contact sales@mattermost.com.

Reviewer: ```suggestion
If you are a reseller, have a non-standard contract, or want to adjust the number of active users on your license during the license period, please contact sales@mattermost.com.
```

System Admins will be alerted 60 days prior to the license expiry via a banner in Mattermost. Select **Renew license now** to start the renewal process in the Customer Portal. You can also dismiss the banner and renew your license at a later date via **System Console > Edition and License**.

> The renewal page in the Customer Portal lists your license information and contact details which is pre-populated based on your existing license subscription. 

Reviewer 2: ```suggestion
The renewal page in the Customer Portal lists your license information and contact details which are pre-populated based on your existing license subscription. 
```

> The renewal page in the Customer Portal lists your license information and contact details which are pre-populated based on your existing license subscription. 

Reviewer: ```suggestion
Once you click **Renew license now**, you are taken to the renewal page in the Customer Portal, which lists your license information and contact details, pre-populated based on your existing license subscription. 
```

Reviewer: Proposing adjustment, to make it clear that "Renew license now" takes the admin to the renewal page. Currently this is not obvious.

**To process your license renewal:**

> 1. Enter your contact and billing information.

Reviewer: Let's match these fields with the renewal form, e.g.

```
1. Enter your **Account Details**, **Additional Contact** and **Payment Details**.
```

Author: ```suggestion
1. Enter your **Account Details**, **Additional Contact**, and **Payment Details**.
```

> 2. Confirm the Mattermost Enterprise Edition version.

Reviewer: ```suggestion
2. Confirm the `Mattermost Enterprise Edition <https://mattermost.com/pricing-self-managed/>_`.
```

3. Confirm the listed number of active users is correct.
4. Select **Complete purchase**. 

> An email with the new license key and information on how to upload the license in the System Console will be sent to the primary email address.

Reviewer 2: Question: who's primary email address? The System Admin's? (If a person can change the primary email address within Mattermost, it may be valuable to include location details).

Author: That's a good question - @jasonblais?

Reviewer: It's the contact entered in the renewal form.

Author: To confirm; there's an account details field and then additional contact field. They're both pre-filled but also editable from what I can see, so they can enter any address and the next time they renew they can change that address should it have changed?

Reviewer: Only the Account Details and the license information (edition + # of licensed seats) are pre-populated, and are based on the email associated with the original license purchase.

And yes, all fields can be edited as needed.

Frequently Asked Questions
~~~~~~~~~~~~~~~~~~~~~~~~~~

How do I renew my license if I don't have internet access?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> If you don't have access to the internet, we'll assist you via email.

Author: ```suggestion
If you don't have access to the internet, please email support@mattermost.com for assistance.
```

Reviewer 2: Clickable email address?

Renewing an Enterprise license
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

You can renew your Mattermost license with a credit card if you have a standard Mattermost contract. When you renew your license, you can also increase the number of active users.

If you are a reseller, have a non-standard contract, or want to adjust the number of active users on your license please contact sales@mattermost.com.

> System Admins will be alerted 60 days prior to the license expiry via a banner in Mattermost. Select **Renew license now** to start the renewal process in the Customer Portal. You can also dismiss the banner and renew your license at a later date via **System Console > Edition and License**.

Author: We mention in the FAQ that the banner returns on browser refresh - @jasonblais, is this still accurate? And what about non-browser based users?

Reviewer: This should still be true, but you can confirm with Steve Mudie (QA). It should work the same for desktop app users, but no such banner is shown on the mobile apps.

Author: @stevemudie, does the banner still behave as mentioned in the docs - i.e., can be closed but then reappears after refresh?

Reviewer: We should document (or otherwise clarify) that this new flow is only applicable to v5.32 and later. Perhaps we can say that for those servers that have not upgraded to v5.32 can contact support to renew their license.

Reviewer 3: > We mention in the FAQ that the banner returns on browser refresh - @jasonblais, is this still accurate? And what about non-browser based users?

This is correct, the banner is dismissible but will come back when they refresh (or reload in the desktop app) within the 60 days before it expires. However, once expired, the banner is no longer dismissible and after the 10 days after expiration, enterprise features are disabled and the banner is changed to: 

<img width="1240" alt="Screen Shot 2021-01-26 at 7 32 56 PM" src="https://user-images.githubusercontent.com/936315/105925002-5e6d6100-600d-11eb-9826-5ed2cad3f1a8.png">

Reviewer 3: > We have [FAQs around that](https://docs.mattermost.com/overview/license-and-subscription.html#what-happens-to-my-license-if-i-don-t-renew-in-time) which also list the features that are affected. LMK if we need more. :)

Perfect, didn't know that was there.

The renewal page in the Customer Portal lists your license information and contact details which is pre-populated based on your existing license subscription. 

> **To process your license renewal:**

Author: @jasonblais, I'm going to remove this in favor of a GIF once the test server is available.

Reviewer: It may be helpful to have the steps followed by the GIF - that way they have a quick checklist of sorts when completing the renewal process.

1. Enter your contact and billing information.
2. Confirm the Mattermost Enterprise Edition version.
3. Confirm the listed number of active users is correct.
4. Select **Complete purchase**. 

An email with the new license key and information on how to upload the license in the System Console will be sent to the primary email address.

> As a final step you can optionally, you can create a Customer Portal account to manage your license subscriptions. This account is linked to your license and not your usrname or user ID, and can be accessed by other Mattermost System Admins.

Author: Confirming whether this is accurate.

Reviewer: This particular part didn't make the first release, but is planned as a follow-up improvement in the near term.

> As a final step you can optionally, you can create a Customer Portal account to manage your license subscriptions. This account is linked to your license and not your usrname or user ID, and can be accessed by other Mattermost System Admins.

Author: ```suggestion
As a final step you can optionally, you can create a Customer Portal account to manage your license subscriptions.
```

Frequently Asked Questions
~~~~~~~~~~~~~~~~~~~~~~~~~~

How do I renew my license if I don't have internet access?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> If you don't have access to the internet, we'll assist you via email.

Author: Is there a specific email address we can provide, or sales@mattermost.com? @jasonblais

Reviewer: `support@mattermost.com`

Renewing an Enterprise license
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> You can renew your Mattermost license with a credit card if you have a standard Mattermost contract. When you renew your license, you can also increase the number of active users.

Author: ```suggestion
You can renew your self-managed Mattermost Enterprise license with a credit card if you have a standard Mattermost contract. When you renew your license, you can also increase the number of active users.

If you're a Mattermost Cloud customer, your subscription auto-renews so you don't need to follow this process.
```

Author: Out of interest, how do Mattermost Cloud customers change the parameters of their subscription? We mention that for self-managed and should mention it for Mattermost Cloud, for consistency.

Reviewer 3: After they have upgraded to Cloud Professional there aren't any more options they can change to unless they want to cancel for which they currently have to contact support to do. Not sure if that answers your question.

Renewing an Enterprise license
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> You can renew your self-managed Mattermost Enterprise license with a credit card if you have a standard Mattermost contract. When you renew your license, you can also increase the number of active users.

Author: ```suggestion
From v5.32 onwards you can renew your self-managed Mattermost Enterprise license with a credit card if you have a standard Mattermost contract. When you renew your license, you can also increase the number of active users.

If you have not upgraded to v5.32 you can contact Support (support@mattermost.com) to renew your license.
```

Reviewer: Propose - `For Mattermost Server version 5.32 and later` for consistency with other pages, e.g. https://docs.mattermost.com/deployment/advanced-permissions.html

![image](https://user-images.githubusercontent.com/13119842/106170845-e8790f00-615e-11eb-9e0e-64df350f537a.png)

> If you're a Mattermost Cloud customer, your subscription auto-renews so you don't need to follow this process.

Author: Out of interest, how do Mattermost Cloud customers change the parameters of their subscription? We mention it for self-hosted and should mention it for Mattermost Cloud for consistency.

Reviewer: The parameters (e.g. users) are automatically updated for them.
